torchmultimodal/modules/layers/quantization.py

```diff
@@ -24,24 +24,60 @@ class Quantization(nn.Module):
     Vector quantization was introduced in Oord et al. 2017 (https://arxiv.org/pdf/1711.00937.pdf)
     to generate high-fidelity images, videos, and audio data.
 
+    The embedding weights are trained with exponential moving average updates as described
+    in original paper.
+
     Args:
         num_embeddings (int): the number of vectors in the embedding space
         embedding_dim (int): the dimensionality of the embedding vectors
 
     Inputs:
-        x (Tensor): Tensor containing a batch of encoder outputs.
+        z (Tensor): Tensor containing a batch of encoder outputs.
                     Expects dimensions to be batch x channel x n dims.
     """
 
-    def __init__(self, num_embeddings: int, embedding_dim: int):
+    def __init__(
+        self,
+        num_embeddings: int,
+        embedding_dim: int,
+        decay: float = 0.99,
+        epsilon: float = 1e-7,
+    ):
         super().__init__()
         self.embedding_dim = embedding_dim
         self.num_embeddings = num_embeddings
 
         self.embedding = nn.Embedding(self.num_embeddings, self.embedding_dim)
-        self.embedding.weight.data.uniform_(
-            -1 / self.num_embeddings, 1 / self.num_embeddings
-        )
+        # These are used in EMA update of embedding weights as m, N, and lambda, respectively, from Oord et al.
+        self._code_avg = self.embedding.weight.detach().clone()
+        self._code_usage = torch.zeros(self.num_embeddings)
+        self._decay = decay
+        # Used in Laplace smoothing of code usage
+        self._epsilon = epsilon
+
+        # Flag to track if we need to initialize embedding with encoder output
+        self._is_embedding_init = False
```

Comment:
I was following the convention of naming boolean flags with "is***", but let me know if aligning with MUGEN's flag name is more important to avoid confusion

Reply:
I wasn't referring to naming of the variable but the default value.

Reply:
Ok, I see. MUGEN was using `_need_init` which is the opposite to `_is_embedding_init`.